symengine/eval.cpp

```diff
@@ -0,0 +1,42 @@
+#include <symengine/eval.h>
+#include <symengine/eval_double.h>
+#ifdef SYMENGINE_HAVE_MPFR
+#include <mpfr.h>
+#include <symengine/real_mpfr.h>
+#include <symengine/eval_mpfr.h>
+#endif // SYMENGINE_HAVE_MPFR
+
+#ifdef SYMENGINE_HAVE_MPC
+#include <mpc.h>
+#include <symengine/complex_mpc.h>
+#include <symengine/eval_mpc.h>
+#endif // SYMENGINE_HAVE_MPC
+
+namespace SymEngine
+{
+
+RCP<const Number> eval(const Basic &b, unsigned long bits, bool real)
+{
+    if (bits <= 53 && real) { // double
+        double d = eval_double(b);
+        return make_rcp<RealDouble>(std::move(d));
+    } else if (bits <= 53 && !real) { // complex double
+        std::complex<double> d = eval_complex_double(b);
+        return make_rcp<ComplexDouble>(std::move(d));
+    } else if (bits > 53 && real) { // mpfr
+#ifdef HAVE_SYMENGINE_MPFR
+        mpfr_class mc = mpfr_class(bits);
+        mpfr_ptr result = mc.get_mpfr_t();
+        eval_mpfr(result, b, MPFR_RNDN);
+        return make_rcp<RealMPFR>(std::move(mc));
+#endif       // HAVE_SYMENGINE_MPFR
```

Comment:
throw an error here if `HAVE_SYMENGINE_MPFR` is not defined saying something like, https://github.com/symengine/symengine.py/blob/master/symengine/lib/symengine_wrapper.pyx#L2111